mllib/src/main/scala/org/apache/spark/ml/feature/FeatureHasher.scala

```diff
@@ -208,7 +208,8 @@ class FeatureHasher(@Since("2.3.0") override val uid: String) extends Transforme
       require(dataType.isInstanceOf[NumericType] ||
```

Comment:
Another option would be to use SchemaUtils `checkColumnTypes` here -- what do you think?

Reply:
I couldn't do that because `NumericType` is an `AbstractDataType` and not a `DataType` so I couldn't use that method.